grudge/trace_pair.py

```diff
@@ -297,7 +310,8 @@ def finish(self):
                          exterior=swapped_remote_dof_array)
 
 
-def _cross_rank_trace_pairs_scalar_field(dcoll, vec, tag=None):
+def _cross_rank_trace_pairs_scalar_field(
+        dcoll: DiscretizationCollection, vec, tag=None) -> list:
```

Comment:
@thomasgibson FYI these could be `List[TracePair]`. (I won't let this hold up the merge.)